### PR TITLE
Add file upload/download documentation to readme

### PR DESCRIPTION
**Pull request by Shane32**
See: #307

### CODE REVIEW

**Comment by Shane32** [comment]
I may suggest we expand this into a sample program (including tests).  We may also want to consider including multipart file decoding as a built-in feature.

File: README.md [hunk 1]
```diff
@@ -764,6 +764,29 @@ are rejected over HTTP GET connections.  Derive from `GraphQLHttpMiddleware` and
 
 As would be expected, subscription requests are only allowed over WebSocket channels.
 
+### File uploading/downloading
+
+A common question is how to upload or download files attached to GraphQL data.
+For instance, storing and retrieving photographs attached to product data.
+
+One common technique is to encode the data as base-64 and transmitting as a custom
```
Comment (sungam3r) [inline README.md:772]:
```suggestion
One common technique is to encode the data as Base64 and transmitting as a custom
```

File: README.md [hunk 2]
```diff
@@ -764,6 +764,29 @@ are rejected over HTTP GET connections.  Derive from `GraphQLHttpMiddleware` and
...
+GraphQL scalar (encoded as a string value within the JSON transport).
+This may not be ideal, but works well for smaller files.  It can also couple with
+response compression (details listed above) to reduce the impact of the base-64
```
Comment (sungam3r) [inline README.md:775]:
```suggestion
response compression (details listed above) to reduce the impact of the Base64
```

File: README.md [hunk 3]
```diff
@@ -764,6 +764,29 @@ are rejected over HTTP GET connections.  Derive from `GraphQLHttpMiddleware` and
...
+encoding.
+
+Another technique is to get or store the data out-of-band.  For responses, this can
+be as simple as a Uri pointing to a location to retrieve the data, especially if
+the data are photographs used in a SPA client application.  This may have additional
+security complications, espeically when used with JWT bearer authentication.
```
Comment (sungam3r) [inline README.md:781]:
```suggestion
security complications, especially when used with JWT bearer authentication.
```

File: README.md [hunk 4]
```diff
@@ -764,6 +764,29 @@ are rejected over HTTP GET connections.  Derive from `GraphQLHttpMiddleware` and
...
+This answer often works well for GraphQL queries, but may not be desired during
+uploads (mutations).
+
+An option for uploading is to upload file data alongside a mutation with the `multipart/form-data`
+content type.  Please see [Issue 307](https://github.com/graphql-dotnet/server/issues/307) and
+[FileUploadTests.cs](https://github.com/graphql-dotnet/server/blob/develop/tests/Transports.AspNetCore.Tests/Middleware/FileUploadTests.cs)
```
Comment (sungam3r) [inline README.md:787]:
```suggestion
[FileUploadTests.cs](https://github.com/graphql-dotnet/server/blob/master/tests/Transports.AspNetCore.Tests/Middleware/FileUploadTests.cs)
```

**Review by sungam3r — APPROVED** [review]

**Pull request merged**